pwndbg/gdblib/heap/__init__.py

```diff
@@ -124,3 +124,48 @@ def resolve_heap(is_first_run: bool = False) -> None:
             )
     else:
         current = pwndbg.gdblib.heap.ptmalloc.DebugSymsHeap()
+
+
+def heap_freebin_address_lookup(addr: int) -> str | None:
+    """
+    Return a string representing the free bin that an address belongs to.
+    None if doesn't belong to a bin.
+
+    Only supports GLIBC.
+    """
+    allocator = pwndbg.gdblib.heap.current
+    if not isinstance(allocator, pwndbg.gdblib.heap.ptmalloc.GlibcMemoryAllocator):
+        return None
+
+    try:
+        heap = pwndbg.gdblib.heap.ptmalloc.Heap(addr)
+    except Exception:
+        return None
+
+    if heap.arena is None:
+        return None
+
+    for chunk in heap:
+        if addr in chunk:
+            arena = chunk.arena
+            if arena:
+                if chunk.is_top_chunk:
+                    return "top chunk"
+
+            if not chunk.is_top_chunk and arena:
+                bins_list = [
+                    allocator.fastbins(arena.address),
```

Comment:
I know this is duplicated from the `vis-heap-chunks` command, but I feel like Martin Fowler/Don Roberts rule of three applies.